ext/sidecar.c

```diff
@@ -22,6 +23,14 @@ ddog_Endpoint *ddtrace_endpoint;
 struct ddog_InstanceId *ddtrace_sidecar_instance_id;
 static uint8_t dd_sidecar_formatted_session_id[36];
 
+#ifdef ZTS
+static __thread zend_string *last_service_name;
+static __thread zend_string *last_env_name;
+#else
+static zend_string *last_service_name;
+static zend_string *last_env_name;
+#endif
```

Comment:
I'm not too happy with that, by now we duplicate the last service name quite a bit. Can we just use `DDTRACE_G(last_flushed_root_service_name)` and `DDTRACE_G(last_flushed_root_env_name)` and (maybe with a rename to `active_root_service_name` or something like that) just use that everywhere.

One question: do you actually need to access that one outside the request lifecycle or why did you make it persistent?

Reply:
Can we not have it dependent on the success of ddog_sidecar_set_universal_service_tags? This should really never fail unless the sidecar connection is fucked, at which point it doesn't matter anyway.